### ISSUE TITLE
Issues with NixOS 15.09 installation on a Hyper-V Generation 2 VM on Windows 10 64-Bit

### ISSUE THREAD

**Issue by lewisl9029** [NixOS/nixpkgs#13971]
## Basic info

To make sure that we are on the same page:
- Kernel: (run `uname -a`)
- System: (NixOS: `nixos-version`, Ubuntu/Fedora: `lsb_release -a`, ...)
- Nix version: (run `nix-env --version`)
- Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`)
## Describe your issue here

This is an installation issue so the fields above weren't quite applicable. 

I am trying to install NixOS 15.09 (15.09.1113.23395c4-x86_64) on a Generation 2 (UEFI-based) Hyper-V VM on Windows 10 64-bit v1511, but running into several issues. See below for details and repro steps. Happy to provide more info if needed.
### Expected result

Should successfully boot into LiveCD. 
VM should be responsive to keyboard input.
### Actual result

Runs into a similar issue as described in https://github.com/NixOS/nixpkgs/issues/9899:

```
waiting for device /dev/disk/by-uuid/... to appear...
Time out waiting for device /dev/disk/by-uuid/..., trying to mount anyway.
mounting /dev/disk/by-uuid/... on /...
mount: mounting /dev/disk/by-uuid... on /mnt/root/ failed: No such file or directory

An error occurred in stage 1 of the boot process, which must mount the root filesystem on /mnt-root and then start stage 2. Press one of the following keys:

i) to launch an interactive shell
f) to start an interactive shell having pid 1 (needed if you want to start stage 2's init manually)
r) to reboot immediately
*) to ignore the error and continue
```

In addition, the VM doesn't respond to any keyboard input through the Hyper-V VM connection at this point, so I can't actually debug it through a shell either.

Neither of these issues exist when installing on a Generation 1 Hyper-V VM.
### Steps to reproduce
1. Create a Generation 2 Hyper-V VM (Not reproducible on a Generation 1 VM, but https://github.com/NixOS/nixpkgs/issues/5872 is still a problem)
2. Disable Secure Boot in VM settings
3. Mount installation ISO, boot the VM and boot into the LiveCD

**Comment by jgillich** [comment]
Unless I'm missing something, the required kernel modules still aren't loaded by default, so you'd have to build your own iso.

https://github.com/NixOS/nixpkgs/issues/9899#issuecomment-162539673

**Comment by lewisl9029** [comment]
Yes I did come across that issue as well, but I suspect it was for Generation 1 Hyper-V VMs (since they mentioned Azure). 

I have tried installing on a Generation 1 VM as well, and did not run into these issues, so I assumed https://github.com/NixOS/nixpkgs/issues/9899 has been fixed, and that this is probably a separate issue (that might need a different set of modules to work around?).

**Comment by jgillich** [comment]
I see. After reading [this](https://gist.github.com/evaryont/c18a166197f1f1a03c74), I'd say Linux doesn't work very well on Gen2 in general

**Comment by taku0** [comment]
Made a fix for this #22455.
